doc/whats-new.rst

```diff
@@ -40,6 +40,8 @@ Enhancements
 Bug fixes
 ~~~~~~~~~
 
+- Bug fix for interpolation with an datetime array. (:issue:`2668`)
```

Comment:
I think @dcherian caught this bug before it made it into a release, so maybe we do not need a bug-fix note?